File: frontend/src/types/dedup-review.ts
export type DuplicateReviewItem = {
  candidate_id: number;
  review_status: string;
  match_type: string;
  confidence_score: string | number;
  comparison_snapshot: Record<string, unknown>;
  source_record: {
    source_record_id: number;
    import_batch_id: number;
    batch_no: string;
    data_source_code: string;
    data_source_name: string;
    source_record_key: string;
    parse_status: string;
    source_stem_text: string;
    source_answer_text?: string | null;
    normalized_question_id?: number | null;
  };
  candidate_question: {
    question_id: number;
    stem_text: string;
    answer_text?: string | null;
  };
  reviewer_name?: string | null;
  reviewed_at?: string | null;
};

export type ReviewDecisionResponse = {
  message: string;
  source_record_id: number;
  normalized_question_id: number;
  parse_status: string;
  review_status: string;
};

export type BulkApproveDuplicateRequest = {
  reviewed_by_user_id: number;
  similarity_threshold: number;
};

export type BulkApproveDuplicateResponse = {
  message: string;
  similarity_threshold: number;
  matched_candidate_count: number;
  approved_source_record_count: number;
};

File: backend/app/schemas/dedup_review.py
from __future__ import annotations

from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, Field


class ReviewCandidateQuestion(BaseModel):
    question_id: int
    stem_text: str
    answer_text: str | None = None


class ReviewSourceRecordSummary(BaseModel):
    source_record_id: int
    import_batch_id: int
    batch_no: str
    data_source_code: str
    data_source_name: str
    source_record_key: str
    parse_status: str
    source_stem_text: str
    source_answer_text: str | None = None
    normalized_question_id: int | None = None


class DuplicateReviewItem(BaseModel):
    candidate_id: int
    review_status: str
    match_type: str
    confidence_score: Decimal
    comparison_snapshot: dict
    source_record: ReviewSourceRecordSummary
    candidate_question: ReviewCandidateQuestion
    reviewer_name: str | None = None
    reviewed_at: datetime | None = None


class DuplicateReviewDecisionRequest(BaseModel):
    reviewed_by_user_id: int


class DuplicateReviewDecisionResponse(BaseModel):
    message: str
    source_record_id: int
    normalized_question_id: int
    parse_status: str
    review_status: str


class BulkApproveDuplicateRequest(BaseModel):
    reviewed_by_user_id: int
    similarity_threshold: float = Field(ge=0, le=1)


class BulkApproveDuplicateResponse(BaseModel):
    message: str
    similarity_threshold: float
    matched_candidate_count: int
    approved_source_record_count: int
